3696/1/top_gems.rb

```diff
@@ -0,0 +1,66 @@
+require 'yaml'
+require 'open-uri'
+require 'optparse'
+require './table_drawer'
+require './scraper'
+require './name_to_link_transformer.rb'
+
+class TopGems
+  def initialize
+    @options = parse
+  end
+
+  def run
+    TableDrawer.new(ranking(gem_names, name_pattern), top).draw
+  end
+
+  private
+
+  attr_reader :options
+
+  def ranking(file, name)
+    file.each_with_object({}) do |str, hsh|
+      next unless str.start_with? name
+
+      transformer_link = NameToLinkTransformer.new(str).link
+      hsh[str] = scrape_value(transformer_link) unless transformer_link == 'no-link'
```

Comment:
Feature request: 
I want to see additional line kinda
`90 of 100 gems are processed failed to find <list of gems here>`
or just 100 `90 of 100 gems are processed` if it's all right

Reply:
Better use symbol here.